sympy/solvers/ode/hypergeometric.py

```diff
@@ -0,0 +1,249 @@
+r'''
+This File contains all the matching and solution method for hypergeometric solver. Main reference
+for this algorithm is taken from  "Non-Liouvillian solutions for second order linear ODEs" by
+L. Chan, E.S. Cheb-Terrab. Although the paper describes an algorithm that should find all solutions
+with 2F1, 0F1 and 1F1,but Sympy only has 2F1 right now.
+The code could be extended to handle 0F1 and 1F1 as described in the paper.
+'''
```

Comment:
This is not well written. Capitalisation is inconsistent. There should be a space after a comma.
```
This module contains the implementation of the 2nd_hypergeometric hint for
dsolve. This is an incomplete implementation of the algorithm described in [1].
The algorithm solves 2nd order linear ODEs of the form

.. math:: y'' + A(x) y' + B(x) y = 0\text{,}

where `A` and `B` are rational functions. The algorithm should find any
solution of the form

.. math:: y = P(x) _pF_q(..; ..;\frac{\alpha x^k + \beta}{\gamma x^k + \delta})\text{,}

where pFq is any of 2F1, 1F1 or 0F1 and `P` is an "arbitrary function".
Currently only the 2F1 case is implemented in SymPy but the other cases are
described in the paper and could be implemented in future (contributions
welcome!).

References
==========

.. [1] L. Chan, E.S. Cheb-Terrab, Non-Liouvillian solutions for second order
       linear ODEs, (2004).
       https://arxiv.org/abs/math-ph/0402063
```